### PR TITLE
Initial base docker image builder, ci build/push

### PR DESCRIPTION
**Pull request by diginc**
We could use a generic base / internal image builder project so I renamed this repo and tested the circle deployment on my personal accounts.  This repo was just the pihole debian base prior but I merged FTL's docker folder in since those appeared to be manually built and we have a need to add a new Docker build env to FTL for docker+armel https://github.com/pi-hole/docker-pi-hole/pull/506.

Will require the DOCKERHUB_* variables be setup on circleci docker-base-images repo.

https://hub.docker.com/r/diginc/ftl-build/tags

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+
+.job_template: &job_template
+  machine:
+    enabled: true
+  steps:
+    - checkout
+    - run:
+        command: |
+          cd "${BASEDIR:-.}/${CIRCLE_JOB}"
+          [[ $IMAGE != *"/"* ]] && IMAGE="diginc/${IMAGE}"
```
Comment (dschaper) [inline .circleci/config.yml]: Should this `IMAGE` point to the Pi-hole namespace?

Comment (diginc) [reply .circleci/config.yml]: I'm afraid my docker hub token being used will cause this to actually deploy over the current prod ftl-build images (I think I have deploy rights there).  

I'm testing the diginc/ftl-build images now to know if that is safe or not.

Comment (dschaper) [reply .circleci/config.yml]: Okay, I can set up pralor with a hub credential and give that bot user a build team access.

**Comment by dschaper** [comment]
Feel free to merge as needed.

**Comment by diginc** [comment]
All the FTL compiles are green so it should be safe to re-release the ftl-build images with this.

Working on adding a arm-qemu/native FTL job as the next step.

**Pull request merged**